x-pack/plugin/src/test/resources/rest-api-spec/test/flattened/10_basic.yml

```diff
@@ -154,5 +154,6 @@
 
   - match:  { hits.total.value: 1 }
   - length: { hits.hits: 1 }
-  - length: { hits.hits.0.fields: 1 }
+  - length: { hits.hits.0.fields: 2 }
```

Comment:
@jtibshirani maybe you have an opinion on this. Seems like we currently see all substructure under a mapped flattened fields as a "leaf" object, regardless of what it contains. With the current implementation of fetching unmapped fields we'd be able to descend into those objects and return them. Or the other way round, if we don't do that, we somehow have to mark the mapped "flattened" nodes in "_source" and stop walking the tree there later when retrieving the unmapped values. Not sure what users would expect here typically, so I went with adapting the expectations in the test for now. Happy to discuss though.

Reply:
This approach seems okay to me for now -- in some sense the sub-fields of the flattened field are in fact 'unmapped'. However I agree the behavior of flattened fields is generally tricky, I've made a note on the meta issue (#60985) to think through how to best handle flattened data.